### ISSUE TITLE
False positive for ifs_same_cond when using cfg!()

### ISSUE THREAD

**Issue by cesarb** [Manishearth/rust-clippy#659]
```rust
#![feature(plugin)]
#![plugin(clippy)]

pub fn test() -> u32 {
    if cfg!(feature = "feature1") {
        1
    } else if cfg!(feature = "feature2") {
        2
    } else {
        3
    }
}
```

```
src/lib.rs:7:15: 7:41 warning: this `if` has the same condition as a previous if, #[warn(ifs_same_cond)] on by default
src/lib.rs:7     } else if cfg!(feature = "feature2") {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib.rs:7:15: 7:41 note: in this expansion of cfg!
src/lib.rs:5:8: 5:34 note: same as this
src/lib.rs:5     if cfg!(feature = "feature1") {
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib.rs:5:8: 5:34 note: in this expansion of cfg!
src/lib.rs:7:15: 7:41 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#ifs_same_cond
```

**Comment by mcarton** [comment]
Expression comparison needs a macro check, I guess `cfg!` expands twice to `true` or twice to `false` here.

**Comment by mcarton** [comment]
Thanks for the report.
